$OpenBSD: patch-configure_ac,v 1.3 2021/02/18 13:24:04 sthen Exp $

drop forced -O2
handle libcdada dependency on C++ standard library
only linux requires libnuma for mysql


Index: configure.ac
--- configure.ac.orig
+++ configure.ac
@@ -30,12 +30,6 @@ AC_MSG_RESULT($host_os)
 AC_MSG_CHECKING(hardware)
 AC_MSG_RESULT($host_cpu)
 
-dnl initial checks; tweakings to CFLAGS and LDFLAGS
-dnl see final checks for tweakings to LIBS
-if test "x$ac_cv_prog_gcc" = xyes ; then
-  CFLAGS="-O2 ${CFLAGS}"
-fi
-
 AC_CHECK_FUNCS([strlcpy])
 
 dnl: shared object handling
@@ -406,10 +400,6 @@ AC_ARG_ENABLE(mysql,
     CFLAGS="$CFLAGS $MYSQL_CFLAGS"
     AC_CHECK_HEADERS([mysql.h mysql/mysql.h])
     CFLAGS="$_save_CFLAGS"
-
-    AC_CHECK_LIB([numa], [numa_bind], [], [AC_MSG_ERROR([
-      ERROR: missing libnuma. Requirement for building MySQL.
-    ])])
     ;;
   no)
     AC_MSG_RESULT(no)
@@ -1138,7 +1128,7 @@ AC_LANG_PUSH([C++])
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([],[])],[AC_MSG_RESULT(yes)],[AC_MSG_ERROR([No C++ compiler found])])
 AC_LANG_POP([C++])
 
-LIBS="-lcdada -lstdc++ ${LIBS}"
+LIBS="-lcdada ${CXXLIB} ${LIBS}"
 AS_CASE(["$WITH_EXTERNAL_DEPS"],
   [yes], [],
   [no], [AC_CHECK_LIB([cdada], [cdada_get_ver], [], [AC_MSG_ERROR([Could not find libcdada])])],
